st2common/st2common/validators/api/reactor.py

```diff
@@ -70,4 +74,12 @@ def validate_trigger_parameters(trigger_type_ref, parameters):
     cleaned = util_schema.validate(instance=parameters, schema=parameters_schema,
                                    cls=util_schema.CustomValidator, use_default=True,
                                    allow_default_none=True)
+
+    # Additional validation fron CronTimer
+    if trigger_type_ref == ResourceReference.to_string_reference(SYSTEM_PACK_NAME, 'st2.CronTimer'):
```

Comment:
The fact that this code looks for a specific `trigger_type_ref` and does special handling in code that should have no clue about the specific nature of trigger types is unacceptable.

I guess it is the same as my point in a different comment. This code knows nothing about the nature of this specific trigger_type i.e. `core.st2.CronTimer` - that StackStorm even has such an internal trigger is an internal detail. I do not think we should be validating that here - given the decoupled nature of the system it should only be handled in RulesEngine. Yes this means rule will be create and lead to a 200 even though trigger parameters might be bad.